source/grid/tria.cc

```diff
@@ -8995,28 +8995,79 @@ Triangulation (const Triangulation<dim, spacedim> &other)
 
 
 
+#ifdef DEAL_II_WITH_CXX11
+template <int dim, int spacedim>
+Triangulation<dim, spacedim>::
+Triangulation (Triangulation<dim, spacedim> &&tria)
+  :
+  Subscriptor(static_cast<Subscriptor&&>(tria)),
+  smooth_grid(tria.smooth_grid),
+  periodic_face_pairs_level_0(tria.periodic_face_pairs_level_0),
+  periodic_face_map(tria.periodic_face_map),
+  levels(tria.levels),
+  faces(tria.faces),
+  vertices(std::move(tria.vertices)),
+  vertices_used(tria.vertices_used),
```

Comment:
I wrote this a few weeks ago and forgot why I had that. Without it, the check that the moved triangulation is empty afterwards fails because it still has vertices. I read up on this some more today; apparently the `std::move` is necessary, because it converts an r-value reference to an x-value. This makes me think that all the other members that are of type `std::vector` should be moved as well. I'll experiment with this and try it on periodic triangulations or triangulations with manifold boundaries to be sure that it works right in all other cases. Could you relabel it as work-in-progress for now?